opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -744,13 +745,13 @@ def __init__(
         parent: Optional[trace_api.SpanContext] = None,
         sampler: Optional[sampling.Sampler] = None,
         trace_config: None = None,  # TODO
-        resource: Resource = None,
+        resource: Optional[Resource] = None,
         attributes: types.Attributes = None,
-        events: Sequence[Event] = None,
+        events: Optional[Sequence[Event]] = None,
         links: Sequence[trace_api.Link] = (),
```

Comment:
Should this be Optional too then?

Reply:
This one is a bit weird, since one of the parent classes (`ReadableSpan`) does not declare it as Optional, so changing it here would lead to a mypy failure instead.